File: model/src/main/java/com/bnorm/barkeep/model/HasId.java
package com.bnorm.barkeep.model;

import java.util.Comparator;

import javax.annotation.Nullable;

import com.fasterxml.jackson.annotation.JsonView;

public interface HasId {

  Comparator<HasId> COMPARATOR = Comparator.comparing(HasId::getId);

  @JsonView(HasId.class)
  @Nullable
  Long getId();
}

File: model/src/main/kotlin/com/bnorm/barkeep/model/HasId.kt
package com.bnorm.barkeep.model

import com.fasterxml.jackson.annotation.JsonView
import java.util.Comparator
import java.util.function.Function

interface HasId {

  @get:JsonView(HasId::class)
  val id: Long

  companion object {

    internal val COMPARATOR = Comparator.comparing(Function<HasId, Long> { it.id })
  }
}
